src/main/java/de/retest/web/DefaultValuesProvider.java

```diff
@@ -0,0 +1,82 @@
+package de.retest.web;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import de.retest.ui.DefaultValueFinder;
+import de.retest.ui.descriptors.IdentifyingAttributes;
+
+public class DefaultValuesProvider implements DefaultValueFinder {
+
+	public static final String DEFAULTS_FILE_PATH = "/defaults.yaml";
+
+	private final Map<String, Map<String, String>> defaultValues;
+
+	public DefaultValuesProvider() {
+		try ( final InputStream url = getClass().getResourceAsStream( DEFAULTS_FILE_PATH ) ) {
+			defaultValues = readAttributesConfigFromFile( url );
+		} catch ( final Exception e ) {
+			// TODO Own exception
+			throw new AttributesConfigLoadException( DEFAULTS_FILE_PATH, e );
+		}
+	}
+
+	private Map<String, Map<String, String>> readAttributesConfigFromFile( final InputStream in ) throws IOException {
+		final Map<String, Map<String, String>> defaultValues = new HashMap<>();
+		final ObjectMapper mapper = new ObjectMapper( new YAMLFactory() );
+		final JsonNode jsonNode = mapper.readTree( in );
+		for ( final Iterator<Entry<String, JsonNode>> elements = jsonNode.fields(); elements.hasNext(); ) {
+			final Entry<String, JsonNode> field = elements.next();
+			final Map<String, String> defaults = new HashMap<>();
+			final ArrayNode valuesNode = (ArrayNode) field.getValue();
```

Comment:
Check if this is really an array?

Reply:
In our case it always is. Since this should not be customizable by clients, we can assume this.